src/pages/ManageStaff/ManageStaff.jsx

```diff
@@ -26,41 +30,76 @@ import UserTable from '../../components/UserTable/UserTable';
 const { SUPERADMIN_ROLE, DRIVER_ROLE, ADMIN_ROLE } = AUTH_ROLES.AUTH_ROLES;
 
 const ManageStaff = ({ cookies }) => {
-  const [users, setUsers] = useState([]);
-  const [usersCopy, setUsersCopy] = useState([]);
+  const [displayedUsers, setDisplayedUsers] = useState([]);
+  const [allUsers, setAllUsers] = useState([]);
+  const [adminUsers, setAdminUsers] = useState([]);
+  const [driverUsers, setDriverUsers] = useState([]);
   const [isSuperAdmin, setIsSuperAdmin] = useState(false);
+  const [currFilter, setCurrFilter] = useState('all');
+
+  const fuse = new Fuse(allUsers, {
+    keys: ['firstName', 'lastName', 'email'],
+  });
+
+  const updateDisplay = () => {
+    if (currFilter === 'all') {
+      setDisplayedUsers(allUsers);
+      fuse.setCollection(allUsers);
+    } else if (currFilter === 'admin') {
+      setDisplayedUsers(adminUsers);
+      fuse.setCollection(adminUsers);
+    } else {
+      setDisplayedUsers(driverUsers);
+      fuse.setCollection(driverUsers);
+    }
+  };
 
   const refreshData = async () => {
-    const currentUser = await getUserFromDB();
-    const userId = currentUser.id;
+    const userId = getCurrentUserId();
     const currentUserRole = await cookies.get(cookieKeys.ROLE);
     const { data } = await PNPBackend.get('/users');
 
     if (currentUserRole === SUPERADMIN_ROLE) {
       setIsSuperAdmin(true);
-      setUsers(data.filter(user => user.id !== userId));
-      setUsersCopy(data.filter(user => user.id !== userId));
+      setAllUsers(data.filter(user => user.id !== userId));
+      const driverData = data.filter(d => d.role === DRIVER_ROLE);
+      setDriverUsers(driverData);
+      const adminData = data.filter(
+        d => (d.role === ADMIN_ROLE || d.role === SUPERADMIN_ROLE) && d.id !== userId,
+      );
+      setAdminUsers(adminData);
+      setDisplayedUsers(data.filter(user => user.id !== userId));
     } else {
       const driverData = data.filter(d => d.role === DRIVER_ROLE);
-      setUsers(driverData);
+      setDriverUsers(driverData);
+      setDisplayedUsers(driverData);
```

Comment:
If the user is an admin, make sure to call `setAllUsers(driverData)`, otherwise admins never see anything (since they can't filter)